File: week_2/bubble_sort.py
#bubble sort
def swap(tgt, src):
	temp = tgt
	tgt = src
	src = temp
	return tgt, src

def bbl_srt(lst):
	for i in range(len(lst)):
		for j in range(i + 1, len(lst)):
			if lst[i] > lst[j]:
				lst[i], lst[j] = swap(lst[i], lst[j])
	return lst

arr = [int(i) for i in input("Enter list elements: ").split()]
print("Sorted array is: ", bbl_srt(arr))

File: CPP-Version/Sort/bubble_sort.cpp
#include <iostream>
#include <sstream>
#include <vector>

using namespace std;

vector<int> arr_comprehension()
{
    cout << "Input Array elements: ";
    vector<int> arr;
    string input;
    getline(cin, input);

    stringstream ss(input);
    int num;
    while (ss >> num)
    {
        arr.push_back(num);
    }

    return arr;
}

void swap(int *lf, int *rt)
{
    int temp = *rt;
    *rt = *lf;
    *lf = temp;
}

vector<int> bubble_sort(vector<int> arr)
{
    // not using for loop as we have to insert
    for (int i; i < size(arr); i++)
    {
        for (int j = i + 1; j < size(arr); j++)
        {

            if (arr[i] > arr[j])
            {
                int lf, rt;
                lf = arr[i];
                rt = arr[j];
                swap(&lf, &rt);
                arr[i] = lf;
                arr[j] = rt;
            }
        }
    }
    return arr;
}

int main()
{
    vector<int> arr = arr_comprehension();
    arr = bubble_sort(arr);
    cout << endl
         << "Sorted array as follows" << endl;
    for (int i : arr)
    {
        cout << i << "\t";
    }
}